### PR TITLE
Auto closing overlays when user scroll or do DnD

### PR DESCRIPTION
**Pull request by ivanmjartan**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10343/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12336/ : FAILURE in 9m 59s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10345/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12338/ : FAILURE in 11m 04s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10346/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12339/ : FAILURE in 5m 14s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10348/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12341/ : FAILURE in 12m 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10350/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12343/ : FAILURE in 9m 51s

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx:16]: is `CustomEvent` supported by the Internet Explorer? (if not, we should add also polyfill).
Technically, we still should support IE11 (until v9 release).

File: libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx:53]: Nitpick: `={true}` can be omitted

File: libs/sdk-ui-dashboard/src/presentation/widget/common/WidgetScrollablePanel.tsx
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/common/WidgetScrollablePanel.tsx:22]: Come commented code, what we plan to do with it? There is some ticket number, but it's not clear what should happen.

File: libs/sdk-ui-vis-commons/src/legend/PopUpLegend/RowLegend.tsx
Comment (kandl) [inline libs/sdk-ui-vis-commons/src/legend/PopUpLegend/RowLegend.tsx:47]: Why is this change necessary?

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx:1]: This Provider should likely live next to the Overlay component in `sdk-ui-kit`, not in `sdk-ui-dashboard`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10351/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12344/ : FAILURE in 18m 01s

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (hackerstanislav) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx:16]: Yes in IE11 is supproted but needs to be constructed another ways https://stackoverflow.com/a/37566606

File: libs/sdk-ui-dashboard/src/presentation/widget/common/WidgetScrollablePanel.tsx
Comment (hackerstanislav) [reply libs/sdk-ui-dashboard/src/presentation/widget/common/WidgetScrollablePanel.tsx:22]: I have this ticket on me and will be continue after this will be merged :)

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx:1]: This components has very specific layout that deepens to Dashboard structure. I will not place to UI-Kit since this component is very Dashboard specific.

File: libs/sdk-ui-vis-commons/src/legend/PopUpLegend/RowLegend.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-vis-commons/src/legend/PopUpLegend/RowLegend.tsx:47]: That was bug in vis-common ... legends fire after click widgetSelect by clicking on legend ico

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx:1]: Moved to SDK-UI-KIT

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10356/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12350/ : FAILURE in 10m 33s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10357/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12351/ : FAILURE in 9m 37s

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx]: custom-event-polyfill added to SDK-UI-Kit

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10359/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12353/ : FAILURE in 48s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10360/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12354/ : FAILURE in 15m 54s

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/OverlayProvider.tsx]: "custom-event" package added

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10364/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12357/ : SUCCESS in 22m 07s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10368/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12359/ : SUCCESS in 27m 41s

File: libs/sdk-ui-vis-commons/src/legend/PopUpLegend/LegendDialog.tsx
Comment (kandl) [inline libs/sdk-ui-vis-commons/src/legend/PopUpLegend/LegendDialog.tsx:61]: `={true}` is redundant

File: libs/sdk-ui-kit/src/Overlay/OverlayProvider.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-kit/src/Overlay/OverlayProvider.tsx:34]: Fixed

File: libs/sdk-ui-vis-commons/src/legend/PopUpLegend/LegendDialog.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-vis-commons/src/legend/PopUpLegend/LegendDialog.tsx:61]: fiexed

File: libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx:53]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10370/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12361/ : SUCCESS in 22m 19s

File: libs/sdk-ui-kit/package.json
Comment (kandl) [inline libs/sdk-ui-kit/package.json:99]: packages in package.json should be sorted - custom-event should take a place after copy-to-clipboard :)

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-ui-kit/package.json
Comment (ivanmjartan) [reply libs/sdk-ui-kit/package.json]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10371/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12362/ : SUCCESS in 31m 36s

**Review by hackerstanislav — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3152

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**
